[Live Range Selection] fast/forms/input-text-maxlength.html and fast/forms/input-text-paste-maxlength.html fail

rniwa opened the pull request:
#### 1cc87d66d1203936b7af407579907b711324f683
<pre>
[Live Range Selection] fast/forms/input-text-maxlength.html and fast/forms/input-text-paste-maxlength.html fail
<a href="https://bugs.webkit.org/show_bug.cgi?id=248433">https://bugs.webkit.org/show_bug.cgi?id=248433</a>

Reviewed by Darin Adler.

Use internals functions to get the visible text instead of relying on getSelection().toString()
to include text within input elements.

* LayoutTests/fast/forms/input-text-maxlength.html:
* LayoutTests/fast/forms/input-text-paste-maxlength.html:

Canonical link: <a href="https://commits.webkit.org/257161@main">https://commits.webkit.org/257161@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/0b74973370434208be9ce7bfbf2b97466986e47e

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/97890 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/7117 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/31124 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/107367 "Built successfully") | [✅ 🛠 🧪 win](https://ews-build.webkit.org/#/builders/10/builds/167645 "Built successfully and passed tests") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/7566 "Built successfully") | [✅ 🛠 mac-debug](https://ews-build.webkit.org/#/builders/71/builds/35891 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/90437 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/104004 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/103532 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/78/builds/5693 "Passed tests") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/84508 "Built successfully") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/32791 "") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/87564 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/89360 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/75703 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/1100 "Built successfully") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/73/builds/20783 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/1087 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/22297 "Passed tests") | | 
| [  ~~🛠 🧪 merge~~](https://ews-build.webkit.org/#/builders/74/builds/4909 "Build was cancelled. Recent messages:Cleaned up git repository; Checked out pull request; Verified commit is squashed; Reviewed by Darin Adler; Validated commit message; Compiled WebKit (warnings); 14 flakes 1 failures; layout-tests (exception)") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/5916 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/60/builds/44691 "Passed tests") | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/75/builds/2453 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/2358 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/41697 "Passed tests") | | 
<!--EWS-Status-Bubble-End-->

darinadler commented on LayoutTests/fast/forms/input-text-maxlength.html:
```diff
@@ -25,10 +25,11 @@
-    document.execCommand('SelectAll');
-    return document.getSelection().toString();
+    const element = document.getElementById(id);
+    const shadowRoot = internals.shadowRoot(element);
```
I noticed in other tests you avoided the need for this. But not practical here?

rniwa replied:
Right, here, we specifically want to get the rendered text, not the value returned by input.value. That's separately tested by domValueOf function defined above.

rniwa committed:
[Live Range Selection] fast/forms/input-text-maxlength.html and fast/forms/input-text-paste-maxlength.html fail
https://bugs.webkit.org/show_bug.cgi?id=248433

Reviewed by Darin Adler.

Use internals functions to get the visible text instead of relying on getSelection().toString()
to include text within input elements.

* LayoutTests/fast/forms/input-text-maxlength.html:
* LayoutTests/fast/forms/input-text-paste-maxlength.html:

Canonical link: https://commits.webkit.org/257161@main
